### PR TITLE
Add `...` args to `filter_keys()`

### PR DESCRIPTION
**Pull request by RasmusSkytte**
### Intent
This PR adds `...` args to `filter_keys()` which are passed to `dplyr::inner_join()`.

This makes the use of the function easier. For example, it allows the user to pass `copy = TRUE` to the `inner_join`.

### Approach
N/A

### Known issues
~~This PR also contains the code being merged in~~
* [x] ~~https://github.com/ssi-dk/SCDB/pull/107~~
* [x] ~~https://github.com/ssi-dk/SCDB/pull/108~~

~~Once merged, this PR will be rebased and opened.~~

~~Tests for PostgreSQL are showing as failing.~~
~~This is caused by an update to the Workflows so that previously undetected errors are now being explicitly shown.~~
~~The source of the error is being fixed in https://github.com/ssi-dk/SCDB/pull/98~~

### Checklist

* [x] The PR passes all local unit tests
* [x] I have documented any new features introduced
* [ ] If the PR adds a new feature, please add an entry in `NEWS.md`
* [ ] A reviewer is assigned to this PR

### CODE REVIEW

File: tests/testthat/test-filter_keys.R [hunk 1]
```diff
@@ -35,3 +35,29 @@ test_that("filter_keys() works", {
...
+
+test_that("filter_keys() works with copy = TRUE", {
+  for (conn in get_test_conns()) {
+
+    x <- get_table(conn, "__mtcars")
+
+    filter <- x |>
+      utils::head(10) |>
+      dplyr::select(name) |>
+      dplyr::collect()
```
Comment (LasseEngboChr) [inline tests/testthat/test-filter_keys.R:48]: suggestion
```
dplyr::pull(name)
```
and remove the 'pull' in expect_equal

**Review by LasseEngboChr — APPROVED** [review]
One suggestion with a simplification of the associated test

File: tests/testthat/test-filter_keys.R (hunk 1 above)
Comment (RasmusSkytte) [reply tests/testthat/test-filter_keys.R:48]: Note that `filter` is used twice in the `expect_equal` call, once in the `pull()` call and once in the `filter_keys()` call.

And it needs to be a data.frame in `filter_keys()`

Comment (RasmusSkytte) [reply tests/testthat/test-filter_keys.R:48]: Also, you comment made me realize that we could improve the visible binding of the `filter_keys` tests, so I have done that now.

**Pull request merged**